File: LeetCode75-sols/Array_String/Python/Reverse_Vowels_of_a_String.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = ['a', 'e', 'i', 'o', 'u']
        s2 = ''
        vowel_str = ''
        for i in range(len(s)):
            if s.lower()[i] in vowels:
                vowel_str += s[i]
        vowel_str = vowel_str[::-1]
        cnt = 0
        for i in range(len(s)):
            if s.lower()[i] in vowels:
                s2 += vowel_str[cnt]
                cnt += 1
            else:
                s2 += s[i]
        return s2

File: LeetCode75-sols/Array_String/CPP/Reverse_Vowels_of_a_String.cpp
class Solution {
public:
    string reverseVowels(string s) {
        string s2 = "";
        string vowel_str = "";
        for (int i = 0; i < s.length(); i++) {
            if (checker(s[i])) {
                vowel_str += s[i];
            }
        }
        int cnt = vowel_str.length();
        for (int j = 0; j < s.length(); j++) {
            if (checker(s[j])) {
                s2 += vowel_str[cnt-1];
                cnt--;
            }
            else {
                s2 += s[j];
            }
        }
        return s2;
    }

    bool checker(char s)
    {
        if (s=='a' | s=='e' | s=='i' | s=='o' | s=='u' | s=='A' | s=='E' | s=='I' | s=='O' | s=='U') {
            return true;
        }
        return false;
    }
};